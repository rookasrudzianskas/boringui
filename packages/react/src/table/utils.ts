import React, {ReactNode} from "react";
import {TableCollection} from "@react-types/table";

export const hasPaginationChild = (
  children: ReactNode | undefined,
  child: React.ElementType,
): {hasPagination: boolean; rowsPerPage: number} => {
  const pagination = React.Children.toArray(children).find(
    (item) => React.isValidElement(item) && item.type === child,
  );

  if (React.isValidElement(pagination)) {
    return {
      hasPagination: true,
      rowsPerPage: pagination?.props?.rowsPerPage || 0,
    };
  }

  return {hasPagination: false, rowsPerPage: 0};
};

export const isInfinityScroll = (collection: TableCollection<unknown>) => {
  const bodyProps = collection.body?.props;

  return bodyProps?.loadingState !== undefined && typeof bodyProps?.onLoadMore === "function";
};
