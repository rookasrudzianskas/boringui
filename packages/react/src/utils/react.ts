import * as React from "react";

export function getValidChildren(children: React.ReactNode) {
  return React.Children.toArray(children).filter((child) =>
    React.isValidElement(child),
  ) as React.ReactElement[];
}

export function addChildrenClass(children: React.ReactNode, className: string) {
  const validChildren = getValidChildren(children);

  return validChildren.map((child) => {
    return React.cloneElement(child, {
      className: `${child.props.className} ${className}`,
    });
  });
}
