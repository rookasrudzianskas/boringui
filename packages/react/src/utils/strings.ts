export const kebabCase = (text: string) => {
  return text
    ?.replace?.(/([a-z])([A-Z])/g, "$1-$2")
    ?.replace?.(/\s+/g, "-")
    .toLowerCase?.();
};

export const includes = (text?: string, search?: string[]) => {
  return search?.some((item) => text?.includes(item));
};
