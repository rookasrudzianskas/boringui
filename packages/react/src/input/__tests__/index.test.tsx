import React from "react";
import {mount} from "enzyme";

import {Input} from "../../index";
import {nativeEvent} from "../../../tests/utils";

describe("Input", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Input placeholder="placeholder" />);

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work with different sizes", () => {
    const wrapper = mount(
      <div>
        <Input size="xs" />
        <Input size="sm" />
        <Input size="lg" />
        <Input size="xl" />
        <Input width="50%" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should work with different status", () => {
    const wrapper = mount(
      <div>
        <Input color="secondary" />
        <Input color="success" />
        <Input color="warning" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should be work with label", () => {
    const wrapper = mount(
      <div>
        <Input label="label" />
        <Input labelRight="label" />
        <Input labelLeft="label" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("should be work with content", () => {
    const wrapper = mount(
      <div>
        <Input contentLeft={<span>test-icon</span>} />
        <Input contentRight={<span>test-icon</span>} />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should set input from value", () => {
    let wrapper = mount(<Input initialValue="test" />);
    let input = wrapper.find("input").getDOMNode() as HTMLInputElement;

    expect(input.value).toEqual("test");

    wrapper = mount(<Input value="test2" />);
    input = wrapper.find("input").getDOMNode() as HTMLInputElement;
    expect(input.value).toEqual("test2");

    wrapper.setProps({value: "test3"});
    input = wrapper.find("input").getDOMNode() as HTMLInputElement;
    expect(input.value).toEqual("test3");
  });

  it("should trigger event when input changed", () => {
    let value = "";
    const callback = jest
      .fn()
      .mockImplementation((e: React.ChangeEvent<HTMLInputElement>) => (value = e.target.value));
    const wrapper = mount(<Input onChange={callback} />);

    wrapper
      .find("input")
      .at(0)
      .simulate("change", {target: {value: "test"}});
    expect(callback).toHaveBeenCalled();
    expect(value).toEqual("test");
  });

  it("should ignore event when input disabled", () => {
    const callback = jest.fn();
    const wrapper = mount(<Input disabled onChange={callback} />);

    wrapper
      .find("input")
      .at(0)
      .simulate("change", {target: {value: "test"}});
    expect(callback).not.toHaveBeenCalled();
  });

  it("should ignore event when input readonly", () => {
    const callback = jest.fn();
    const wrapper = mount(<Input readOnly onChange={callback} />);

    wrapper
      .find("input")
      .at(0)
      .simulate("change", {target: {value: "test"}});
    expect(callback).not.toHaveBeenCalled();
  });

  it("should clear text", () => {
    let value = "";
    const callback = jest
      .fn()
      .mockImplementation((e: React.ChangeEvent<HTMLInputElement>) => (value = e.target.value));
    const clearHandler = jest.fn();
    const wrapper = mount(<Input clearable onChange={callback} onClearClick={clearHandler} />);

    wrapper
      .find("input")
      .at(0)
      .simulate("change", {target: {value: "test"}});
    expect(callback).toHaveBeenCalled();
    expect(value).toEqual("test");

    wrapper.find(".nextui-input-clear-button").at(0).simulate("click", nativeEvent);
    expect(clearHandler).toHaveBeenCalled();
    expect(value).toEqual("");
  });

  it("should trigger focus correctly", () => {
    const focus = jest.fn();
    const blur = jest.fn();
    const wrapper = mount(<Input onBlur={blur} onFocus={focus} />);

    const input = wrapper.find("input").at(0);

    input.simulate("focus");
    expect(focus).toHaveBeenCalled();
    input.simulate("blur");
    expect(blur).toHaveBeenCalled();
  });

  it("should trigger content event", () => {
    const click = jest.fn();
    const wrapper = mount(
      <Input
        contentClickable
        contentLeft={<span id="test-icon">icon</span>}
        onContentClick={click}
      />,
    );

    wrapper.find("#test-icon").simulate("click", nativeEvent);
    expect(click).toHaveBeenCalled();
  });

  it("should ignore content event when input is disabled", () => {
    const click = jest.fn();
    const wrapper = mount(
      <Input
        contentClickable
        disabled
        contentLeft={<span id="test-icon">icon</span>}
        onContentClick={click}
      />,
    );

    wrapper.find("#test-icon").simulate("click", nativeEvent);
    expect(click).not.toHaveBeenCalled();
  });

  // check ref is available: https://github.com/geist-org/react/issues/189
  it("should forward ref by default", () => {
    const ref = React.createRef<HTMLInputElement>();
    const wrapper = mount(<Input ref={ref} />);

    expect(ref.current).not.toBeNull();
    expect(() => wrapper.unmount()).not.toThrow();
  });
});
