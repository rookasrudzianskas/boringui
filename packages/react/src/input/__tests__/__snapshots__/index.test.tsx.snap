// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should be work with content 1`] = `"<div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><span class=\\"nextui-c-PJLV nextui-c-PJLV-dBGXHd-applyStyles-true nextui-input-content nextui-input-content--left\\"><span>test-icon</span></span><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-c-jeuecp-bVyrVm-hasLeftContent-true nextui-input nextui-input-left-content\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-25\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-c-jeuecp-bVsTEo-hasRightContent-true nextui-input nextui-input-nextui-input-right-content\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-27\\" value=\\"\\"><span class=\\"nextui-c-PJLV nextui-c-PJLV-dBGXHd-applyStyles-true nextui-input-content nextui-input-content--right\\"><span>test-icon</span></span></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div></div>"`;

exports[`Input should be work with label 1`] = `"<div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><label class=\\"nextui-c-hzQjrs nextui-input-block-label\\" for=\\"react-aria5747947790-19\\" id=\\"react-aria5747947790-20\\">label</label><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><div class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-19\\" aria-labelledby=\\"react-aria5747947790-20\\" value=\\"\\"></div></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-21\\" value=\\"\\"><span class=\\"nextui-c-gieWTI nextui-c-gieWTI-bPbUHf-isRight-true nextui-c-gieWTI-bzgJTZ-cv nextui-input-label--right\\">label</span></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><span class=\\"nextui-c-gieWTI nextui-c-gieWTI-ceZkzL-isRight-false nextui-c-gieWTI-bzgJTZ-cv nextui-input-label--left\\">label</span><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-23\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div></div>"`;

exports[`Input should work with different sizes 1`] = `"<div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-jwvDph-size-xs nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-3\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-iUdBCG-size-sm nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-5\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-7\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gNajeV-size-xl nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-9\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-imosJd-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-11\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div></div>"`;

exports[`Input should work with different status 1`] = `"<div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-kWLfoB-color-secondary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-13\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-gzViYB-color-success nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-15\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div><div class=\\"nextui-c-iwmZdB nextui-c-iwmZdB-jzFoKt-color-warning nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal\\" data-state=\\"normal\\"><div class=\\"nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input\\"><label class=\\"nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow\\"><input aria-multiline=\\"false\\" aria-placeholder=\\"\\" aria-readonly=\\"false\\" aria-required=\\"false\\" autocomplete=\\"off\\" class=\\"nextui-c-jeuecp nextui-input\\" data-state=\\"normal\\" placeholder=\\"\\" type=\\"text\\" id=\\"react-aria5747947790-17\\" value=\\"\\"></label></div><div class=\\"nextui-c-gJvMas nextui-input-helper-text-container\\"></div></div></div>"`;
